from_mwt_ds/DataScience/estimate-notebook-parallel.ipynb

```diff
@@ -0,0 +1,234 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3e3c6305",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create or Attach AML Compute\n",
+    "from azureml.core.compute import ComputeTarget, AmlCompute\n",
+    "from azureml.core.compute_target import ComputeTargetException\n",
+    "\n",
+    "# choose a name for your cluster\n",
+    "cluster_name = \"rlos-cfe-cluster\"\n",
+    "\n",
+    "try:\n",
+    "    compute_target = ComputeTarget(workspace=ws, name=cluster_name)\n",
+    "    print('Found existing compute target')\n",
+    "except ComputeTargetException:\n",
+    "    print('Creating a new compute target...')\n",
+    "    compute_config = AmlCompute.provisioning_configuration(vm_size='STANDARD_D2_V2', max_nodes=4)\n",
+    "\n",
+    "    # create the cluster\n",
+    "    compute_target = ComputeTarget.create(ws, cluster_name, compute_config)\n",
+    "\n",
+    "    # can poll for a minimum number of nodes and for a specific timeout. \n",
+    "    # if no min node count is provided it uses the scale settings for the cluster\n",
+    "    compute_target.wait_for_completion(show_output=True, min_node_count=None, timeout_in_minutes=20)\n",
+    "\n",
+    "# use get_status() to get a detailed status for the current cluster. \n",
+    "print(compute_target.get_status().serialize())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "294d384e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Use simple script\n",
+    "# CP script-run\n",
+    "# cb to script-run\n",
+    "import shutil, os\n",
+    "source_directory = 'script-run'\n",
+    "shutil.rmtree(source_directory)\n",
+    "os.makedirs(source_directory, exist_ok=True)\n",
+    "shutil.copytree('cb', os.path.join(source_directory, 'cb'))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8539b073",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile $source_directory/estimate.py\n",
+    "\n",
+    "# %%run $source_directory/hello.py\n",
+    "\n",
+    "# Copyright (c) Microsoft Corporation. All rights reserved.\n",
+    "# Licensed under the MIT License.\n",
+    "import sys\n",
+    "import os\n",
+    "from mpi4py import MPI\n",
+    "\n",
+    "print(\"*********************************************************\")\n",
+    "print(\"Hello Azure ML!\")\n",
+    "\n",
+    "mounted_input_path = sys.argv[1]\n",
+    "mounted_output_path = sys.argv[2]\n",
+    "\n",
+    "print(\"Argument 1: %s\" % mounted_input_path)\n",
+    "print(\"Argument 2: %s\" % mounted_output_path)\n",
+    "print(\"Number of nodes {}:\".format(MPI.COMM_WORLD.Get_size()))\n",
+    "print(\"Node index {}:\".format(MPI.COMM_WORLD.Get_rank()))\n",
+    "\n",
+    "import pandas as pd\n",
+    "\n",
+    "df = pd.read_csv(os.path.join(mounted_input_path, '01.csv'), parse_dates=['t']).set_index('t')\n",
```

Comment:
it looks like right now both nodes are still processing only 1 file. Can we update it to make different nodes processing different files and then this example should be done.

Reply:
should I use the MPI Index for this ie if Index = 1, use this file, else use other file?